File: plugins/kotlin/idea/tests/testData/decompiler/navigation/userJavaCode/ExtensionProperty.java
import testData.libraries.*;
import java.util.*;

class TestOverload {
    {
        System.out.println(ExtensionPropertyKt.getExProp(""));
        ArrayList<String> p = new ArrayList<>();
        System.out.println(ExtensionPropertyKt.getExProp(p));
        System.out.println(ExtensionPropertyKt.getExPropWithContextReceiver("context", p));
    }
}

File: plugins/kotlin/idea/tests/testData/decompiler/navigation/fromJavaSource/extensionProperty.kt
package testData.libraries

public val String.exProp : String
  get() {
    return this
  }

public val Int.exProp : Int
  get() {
    return this
  }

public val <T> java.util.List<T>.exProp : String
  get() {
    return ""
  }

context(String)
public val <T> java.util.List<T>.exPropWithContextReceiver : String
  get() {
    return ""
  }
